### PR TITLE
NEW: Stock movement pdf & odt document generation

### PR DESCRIPTION
**Pull request by Tim-Otte**
# New: Stock movement pdf & odt document generation

In this PR I fixed the pdf document generation for stock movements and added the odt generation functionality with new custom object tags (_object_description_, _object_nb_products_unique_, _object_nb_products_total_, _object_last_movement_) and line tags (_line_id_, _line_product_ref_, _line_qty_, _line_date_, _line_label_, _line_inventory_code_, _line_price_, _line_author_). Additionally I translated the user rights for stock and stock movement from french to english (**Caution**: This changed a lot of files and modules!). The document models for stock movements can be modified and changed in a new tab in the stock module admin page (The tab title is currently only available in english and german).

### CODE REVIEW

**Comment by Tim-Otte** [comment]
I don't know why the Travis CI build failed (my changes did not affect the UserGroup setup, PR #13681 may fix this issue) but the PR is ready to test and merge. I'd be happy to see this feature in the next update or in v12 of Dolibarr.

File: htdocs/core/modules/modStock.class.php [hunk 1]
```diff
@@ -114,39 +114,39 @@ public function __construct($db)
 		$this->rights_class = 'stock';
 
 		$this->rights[0][0] = 1001;
-		$this->rights[0][1] = 'Lire les stocks';
+		$this->rights[0][1] = 'Read stocks';
 		$this->rights[0][2] = 'r';
 		$this->rights[0][3] = 0;
-		$this->rights[0][4] = 'lire';
+		$this->rights[0][4] = 'read';
```
Comment (fappels) [inline htdocs/core/modules/modStock.class.php:120]: Refactoring rights can cause issues on external modules, at least entry in developer changelog is required. Also rights are in database so migration should be well tested.

Comment (Tim-Otte) [reply htdocs/core/modules/modStock.class.php:120]: Sure, this change is a bit risky for migration, but it is necessary for better code quality. I keep coming across French comments or variables, and I always have to consult a translator to even understand what the section is about.

**Comment by eldy** [comment]
Can you split the PR in 2 (at least on PR for the new feature and the other one for the refactoring of permissions). 
A PR must be atomic. It means it must contains the lower possible changes for 1 need (1 bug fix or 1 new feature) without breaking usability of code. If a PR can be split into several PRs, it often means your PR is not atomic.

**Comment by Tim-Otte** [comment]
Closing this to create atomic pull requests

**Pull request closed without merging**